server/src/test/java/org/opensearch/action/support/replication/TransportWriteActionTests.java

```diff
@@ -395,6 +398,48 @@ public void testReplicaProxy() throws InterruptedException, ExecutionException {
         }
     }
 
+    public void testPrimaryClosedDoesNotFailShard() {
```

Comment:
Why not add a test in corruptionIT as well and ensure that the cluster is green after some time?